src/util.ts

```diff
@@ -26,3 +26,24 @@ export const urlWithContextAsQuery = (url: URL, context: IContext) => {
         });
     return urlWithQuery;
 };
+
+const sortObjectProperties = (
+    obj: Record<string, unknown>
+): Record<string, unknown> => {
+    const sortedKeys = Object.keys(obj).sort();
+    const sortedObj: Record<string, unknown> = {};
+    sortedKeys.forEach((key) => {
+        if (obj[key] !== null && typeof obj[key] === 'object') {
+            sortedObj[key] = sortObjectProperties(
+                obj[key] as Record<string, unknown>
+            );
+        } else {
+            sortedObj[key] = obj[key];
+        }
+    });
+
+    return sortedObj;
+};
```

Comment:
Context has a simple structure. I don't think we need recursion here.
https://docs.getunleash.io/reference/unleash-context#structure

Output doesn't need to directly correspond to context structure.

```ts
const sortEntries = (record: Record<string, string>) => {
    return Object.entries(record).sort(([a], [b]) => a.localeCompare(b));
};

const contextString = (context: IContext): string => {
    const { properties, ...fields } = context;

    return JSON.stringify({
        fields: sortEntries(fields),
        properties: sortEntries(context.properties || {}),
    });
};
```